internal/datastore/common/gc.go

```diff
@@ -111,6 +111,7 @@ func startGarbageCollectorWithMaxElapsedTime(ctx context.Context, gc GarbageColl
 	backoffInterval.InitialInterval = interval
 	backoffInterval.MaxInterval = max(MaxGCInterval, interval)
 	backoffInterval.MaxElapsedTime = maxElapsedTime
+	backoffInterval.Reset()
```

Comment:
This is the other issue I discovered during testing. Without this, the initial interval used by the backoff will be the default `500ms` until `backoffInterval.Reset()` is called. This is an edge case, but it can cause the wrong initial value to be used if the first iteration of the GC has an error.